### PR TITLE
Define `str` and `repr` for client.

### PR DESCRIPTION
**Pull request by vtomole**

### CODE REVIEW

**Comment by vtomole** [comment]
@dowusu-antwi @richrines1 Ready for review.

File: applications_superstaq/superstaq_client.py [hunk 1]
```diff
@@ -387,3 +392,17 @@ def _make_request(self, request: Callable[[], requests.Response]) -> requests.Re
...
+    def __repr__(self) -> str:
+        return f"""applications_superstaq.superstaq_client._SuperstaQClient(
+        remote_host={self.url!r},
+        api_key={self.api_key!r},
+        client_name={self.client_name!r},
+        default_target={self.default_target!r},
+        api_version={self.api_version!r},
+        max_retry_seconds={self.max_retry_seconds!r},
+        verbose={self.verbose!r},
+    )"""
```
Comment (richrines1) [inline applications_superstaq/superstaq_client.py:408]: this indents a little weirdly, e.g.:
```
>>> applications_superstaq.superstaq_client._SuperstaQClient("https://www.example.com", "token", "none")
applications_superstaq.superstaq_client._SuperstaQClient(
            remote_host='https://www.example.com/v0.1.0',
            api_key='token',
            client_name='none',
            default_target=None,
            api_version='v0.1.0',
            max_retry_seconds=3600,
            verbose=False,
        )
```

possible fix:
```suggestion
    def __repr__(self) -> str:
        return textwrap.dedent(
            f"""\
            applications_superstaq.superstaq_client._SuperstaQClient(
                remote_host={self.url!r},
                api_key={self.api_key!r},
                client_name={self.client_name!r},
                default_target={self.default_target!r},
                api_version={self.api_version!r},
                max_retry_seconds={self.max_retry_seconds!r},
                verbose={self.verbose!r},
            )"""
        )
```
which looks like:
```
>>> applications_superstaq.superstaq_client._SuperstaQClient("https://www.example.com", "token", "none")
applications_superstaq.superstaq_client._SuperstaQClient(
    remote_host='https://www.example.com/v0.1.0',
    api_key='token',
    client_name='none',
    default_target=None,
    api_version='v0.1.0',
    max_retry_seconds=3600,
    verbose=False,
)
```

File: applications_superstaq/superstaq_client.py [hunk 2]
```diff
@@ -67,32 +67,37 @@ def __init__(
...
+        ), f"Only api versions {self.SUPPORTED_VERSIONS} are accepted but was {self.api_version}"
         assert (
             default_target is None or default_target in self.SUPPORTED_TARGETS
         ), f"Target can only be one of {self.SUPPORTED_TARGETS} but was {default_target}."
         assert max_retry_seconds >= 0, "Negative retry not possible without time machine."
 
         self.url = f"{url.scheme}://{url.netloc}/{api_version}"
         self.verify_https: bool = (
-            applications_superstaq.API_URL + "/" + applications_superstaq.API_VERSION == self.url
+            applications_superstaq.API_URL + "/" + self.api_version == self.url
```
[the remark this answers is not in the document]
Comment (vtomole) [reply applications_superstaq/superstaq_client.py]: It's not, but better to over annotate than under.

**Review by richrines1 — APPROVED** [review]
lgtm

**Pull request merged**